mlir-tensorrt/compiler/include/mlir-tensorrt/Compiler/Client.h

```diff
@@ -145,8 +172,43 @@ class CompilerClient {
   /// used to create the PM.
   llvm::DenseMap<PassManagerKey, std::unique_ptr<CompilationTaskBase>>
       cachedPassManagers;
+
+  static llvm::ManagedStatic<llvm::StringMap<OptionsConstructorFuncT>> registry;
 };
 
+/// Helper to register option types with the client
+template <typename OptionsT, typename TaskT>
+StatusOr<std::unique_ptr<mlir::OptionsContext>>
+optionsCreateFromArgs(const CompilerClient &client,
+                      const std::vector<llvm::StringRef> &args) {
+  // Load available extensions.
+  mlir::MLIRContext *context = client.getContext();
+  mlir::plan::PlanDialect *planDialect =
+      context->getLoadedDialect<mlir::plan::PlanDialect>();
+  compiler::TaskExtensionRegistry extensions =
+      planDialect->extensionConstructors.getExtensionRegistryForTask<TaskT>();
+
+  auto result = std::make_unique<OptionsT>(std::move(extensions));
+
+  std::string err;
+  if (failed(result->parse(args, err))) {
+    std::string line = llvm::join(args, " ");
+    return getInternalErrorStatus(
+        "failed to parse options string {0} due to error: {1}", line, err);
+  }
+
+  // TODO: Figure out whether to add a method in the base class like
+  // "finalizeOptions" or a callback here, or something else if
+  // `inferDeviceOptionsFromHost` is unique to StableHLO.
+  //
+  // Populate device options from host information.
+  Status inferStatus = result->inferDeviceOptionsFromHost();
+  if (!inferStatus.isOk())
+    return inferStatus;
+
+  return std::unique_ptr<mlir::OptionsContext>(result.release());
```

Comment:
I think you should be able to just `return result` or `return std::move(result)`?

Reply:
It's a pointer to the derived type, so I get:
```
Client.h:197:10: error: no viable conversion from returned value of type 'typename _MakeUniq<StableHLOToExecutableOptions>::__single_object' (aka 'unique_ptr<mlirtrt::compiler::StableHLOToExecutableOptions>') to function return type 'StatusOr<std::unique_ptr<mlir::OptionsContext>>'
```